src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
I may have been incorrect; looking at the AST generated by

```rust
enum TestEnum {
    Arm1(),
    Arm2,
}


fn main() {
    let test = TestEnum::Arm2;
    match test {
        TestEnum::Arm1 => {}
        TestEnum::Arm2 => {}
    }
}
```
[ast.txt](https://github.com/rust-lang-nursery/rustfmt/files/871021/ast.txt),

Arm2 isn't a tuple struct at all, but rather, a path, so that would explain why it doesn't have parens added